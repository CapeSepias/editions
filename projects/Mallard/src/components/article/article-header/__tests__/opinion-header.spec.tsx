import React from 'react'
import TestRenderer, { ReactTestRendererJSON } from 'react-test-renderer'
import { OpinionHeader } from '..'

describe('OpinionHeader', () => {
    it('should show a full featured opinion header', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <OpinionHeader
                image={{ source: 'media', path: 'path/to/image.png' }}
                headline="Theresa May makes final speech as PM"
                standfirst="Rolling coverage of the day’s political developments as they happen"
                byline="Andrew Sparrow"
                kicker="News"
            />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
    it('should show opinion header without an image or a kicker', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <OpinionHeader
                headline="Theresa May makes final speech as PM"
                standfirst="Rolling coverage of the day’s political developments as they happen"
                byline="Andrew Sparrow"
            />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
})
