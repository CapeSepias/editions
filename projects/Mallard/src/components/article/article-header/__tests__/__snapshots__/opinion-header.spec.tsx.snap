// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpinionHeader should show a full featured opinion header 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "paddingBottom": 12,
          "paddingHorizontal": 12,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#999999",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://d2cf1ljtg904cv.cloudfront.net/issue/media/phone/media/path/to/image.png",
        }
      }
      style={
        Array [
          Object {
            "backgroundColor": "rgba(0,0,0,.25)",
            "width": "100%",
          },
          Object {
            "aspectRatio": 1.5,
            "marginBottom": 3,
          },
        ]
      }
    />
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#121212",
              "flexShrink": 0,
              "fontFamily": "GHGuardianHeadline-Regular",
              "fontSize": 28,
              "lineHeight": 30,
              "marginTop": 4,
            },
            Array [
              Object {
                "marginRight": 24,
              },
              Object {
                "color": "#121212",
              },
              Object {},
            ],
          ]
        }
      >
        <RNSVGSvgView
          bbHeight={25}
          bbWidth={38}
          fill="none"
          height={25}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              undefined,
              null,
              Object {
                "flex": 0,
                "height": 25,
                "width": 38,
              },
            ]
          }
          width={38}
        >
          <RNSVGGroup
            fill={null}
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          >
            <RNSVGPath
              d="M8.864.333h7.74c-.932 6.867-1.756 13.6-2.114 21.134H.3C1.59 14.133 4.277 7.2 8.864.333zm18.024 0h7.633c-.825 6.867-1.756 13.6-2.114 21.134H18.252c1.47-7.334 4.05-14.267 8.636-21.134z"
              fill={
                Array [
                  0,
                  4294934287,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={
                Array [
                  "fill",
                ]
              }
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        Theresa May makes final speech as PM
      </Text>
      <Text
        style={
          Array [
            Object {
              "flexShrink": 0,
              "fontFamily": "GuardianTextEgyptian-Reg",
              "fontSize": 17,
              "lineHeight": 21,
            },
            Object {
              "fontFamily": "GuardianTextEgyptian-Bold",
            },
            Array [
              Object {
                "color": "#121212",
              },
              Object {
                "marginBottom": 12,
              },
              Object {
                "color": "#121212",
              },
            ],
          ]
        }
      >
        Andrew Sparrow
      </Text>
    </View>
  </View>,
  <RNSVGSvgView
    bbHeight={16}
    bbWidth="100%"
    fill="none"
    height={16}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        undefined,
        null,
        Object {
          "flex": 0,
          "height": 16,
          "width": "100%",
        },
      ]
    }
    width="100%"
  >
    <RNSVGGroup
      fill={null}
      fillOpacity={1}
      fillRule={1}
      font={Object {}}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={
        Array [
          "fill",
        ]
      }
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={1}
      vectorEffect={0}
      x={0}
      y={0}
    >
      <RNSVGRect
        fill={
          Array [
            0,
            4279374354,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        height={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        rx={0}
        ry={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        width="100%"
        x={0}
        y={0}
      />
      <RNSVGRect
        fill={
          Array [
            0,
            4279374354,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        height={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        rx={0}
        ry={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        width="100%"
        x={0}
        y={4}
      />
      <RNSVGRect
        fill={
          Array [
            0,
            4279374354,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        height={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        rx={0}
        ry={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        width="100%"
        x={0}
        y={8}
      />
      <RNSVGRect
        fill={
          Array [
            0,
            4279374354,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        height={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        rx={0}
        ry={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        width="100%"
        x={0}
        y={12}
      />
    </RNSVGGroup>
  </RNSVGSvgView>,
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "paddingBottom": 12,
          "paddingHorizontal": 12,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#999999",
        },
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "marginBottom": 12,
              "marginTop": 12,
              "paddingTop": 3,
              "width": "100%",
            },
            Object {
              "backgroundColor": "#ffffff",
              "borderColor": "#999999",
            },
            Object {
              "alignContent": "stretch",
              "alignItems": "stretch",
              "justifyContent": "flex-end",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "flexShrink": 0,
                "fontFamily": "GuardianTextEgyptian-Reg",
                "fontSize": 17,
                "lineHeight": 21,
              },
              Array [
                Object {
                  "color": "#121212",
                },
                Object {},
              ],
            ]
          }
        >
          Rolling coverage of the day’s political developments as they happen
        </Text>
      </View>
    </View>
  </View>,
]
`;

exports[`OpinionHeader should show opinion header without an image or a kicker 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "paddingBottom": 12,
          "paddingHorizontal": 12,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#999999",
        },
      ]
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#121212",
              "flexShrink": 0,
              "fontFamily": "GHGuardianHeadline-Regular",
              "fontSize": 28,
              "lineHeight": 30,
              "marginTop": 4,
            },
            Array [
              Object {
                "marginRight": 24,
              },
              Object {
                "color": "#121212",
              },
              Object {},
            ],
          ]
        }
      >
        <RNSVGSvgView
          bbHeight={25}
          bbWidth={38}
          fill="none"
          height={25}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              undefined,
              null,
              Object {
                "flex": 0,
                "height": 25,
                "width": 38,
              },
            ]
          }
          width={38}
        >
          <RNSVGGroup
            fill={null}
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          >
            <RNSVGPath
              d="M8.864.333h7.74c-.932 6.867-1.756 13.6-2.114 21.134H.3C1.59 14.133 4.277 7.2 8.864.333zm18.024 0h7.633c-.825 6.867-1.756 13.6-2.114 21.134H18.252c1.47-7.334 4.05-14.267 8.636-21.134z"
              fill={
                Array [
                  0,
                  4294934287,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={
                Array [
                  "fill",
                ]
              }
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        Theresa May makes final speech as PM
      </Text>
      <Text
        style={
          Array [
            Object {
              "flexShrink": 0,
              "fontFamily": "GuardianTextEgyptian-Reg",
              "fontSize": 17,
              "lineHeight": 21,
            },
            Object {
              "fontFamily": "GuardianTextEgyptian-Bold",
            },
            Array [
              Object {
                "color": "#121212",
              },
              Object {
                "marginBottom": 12,
              },
              Object {
                "color": "#121212",
              },
            ],
          ]
        }
      >
        Andrew Sparrow
      </Text>
    </View>
  </View>,
  <RNSVGSvgView
    bbHeight={16}
    bbWidth="100%"
    fill="none"
    height={16}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        undefined,
        null,
        Object {
          "flex": 0,
          "height": 16,
          "width": "100%",
        },
      ]
    }
    width="100%"
  >
    <RNSVGGroup
      fill={null}
      fillOpacity={1}
      fillRule={1}
      font={Object {}}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={
        Array [
          "fill",
        ]
      }
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={1}
      vectorEffect={0}
      x={0}
      y={0}
    >
      <RNSVGRect
        fill={
          Array [
            0,
            4279374354,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        height={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        rx={0}
        ry={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        width="100%"
        x={0}
        y={0}
      />
      <RNSVGRect
        fill={
          Array [
            0,
            4279374354,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        height={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        rx={0}
        ry={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        width="100%"
        x={0}
        y={4}
      />
      <RNSVGRect
        fill={
          Array [
            0,
            4279374354,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        height={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        rx={0}
        ry={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        width="100%"
        x={0}
        y={8}
      />
      <RNSVGRect
        fill={
          Array [
            0,
            4279374354,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        height={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        rx={0}
        ry={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        width="100%"
        x={0}
        y={12}
      />
    </RNSVGGroup>
  </RNSVGSvgView>,
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "paddingBottom": 12,
          "paddingHorizontal": 12,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#999999",
        },
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "marginBottom": 12,
              "marginTop": 12,
              "paddingTop": 3,
              "width": "100%",
            },
            Object {
              "backgroundColor": "#ffffff",
              "borderColor": "#999999",
            },
            Object {
              "alignContent": "stretch",
              "alignItems": "stretch",
              "justifyContent": "flex-end",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "flexShrink": 0,
                "fontFamily": "GuardianTextEgyptian-Reg",
                "fontSize": 17,
                "lineHeight": 21,
              },
              Array [
                Object {
                  "color": "#121212",
                },
                Object {},
              ],
            ]
          }
        >
          Rolling coverage of the day’s political developments as they happen
        </Text>
      </View>
    </View>
  </View>,
]
`;
